experimental/computeir/Literal.fbs

```diff
@@ -0,0 +1,203 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace org.apache.arrow.computeir.flatbuf;
+
+table ListLiteral {
+  values: [Literal];
+}
+
+table StructLiteral {
+  values: [KeyValue];
+}
+
+table KeyValue {
+  key: string;
+  value: Literal;
+}
+
+table MapLiteral {
+  values: [KeyValue];
```

Comment:
`KeyValue` forces the key to be a string but a map can accept any type as key type, is it deliberate to restrict MapLiteral to string keys?

Reply:
It was, but only because that's the most common case, and we wanted to keep the prototype fairly limited. However, I think we can allow any type here if Arrow allows it.

Reply:
Done.